cmd/sonobuoy/app/args.go

```diff
@@ -359,13 +362,15 @@ func AddPluginListFlag(p *[]string, flags *pflag.FlagSet) {
 	flags.StringSliceVarP(p, "plugin", "p", []string{"e2e", "systemd-logs"}, "Describe which plugin's images to interact with (Valid plugins are 'e2e', 'systemd-logs').")
 }
 
-// AddKubernetesVersionFlag adds a string flag for the user to specify the cluster version to assume instead of reaching
-// out to the cluster to auto-detect it.
-func AddKubernetesVersionFlag(version *string, flags *pflag.FlagSet) {
-	flags.StringVar(
-		version, "kubernetes-version", "",
-		"Version to assume for Kubernetes. If empty, the cluster will be queried for its version",
-	)
+// AddKubernetesVersionFlag initialises an image version flag.
+func AddKubernetesVersionFlag(imageVersion *image.ConformanceImageVersion, flags *pflag.FlagSet) {
+	help := "Use default Conformance image, but override the version. "
+	help += "Default is 'auto', which will be set to your cluster's version if detected, erroring otherwise. "
+	help += "'ignore' will try version resolution but ignore errors. "
+	help += "'latest' will find the latest dev image/version upstream."
+
+	*imageVersion = image.ConformanceImageVersionAuto
+	flags.Var(imageVersion, "kubernetes-version", help)
```

Comment:
Moving towards this flag/naming since this version is more generic than just the conformance image.

The preexisting flag should work as well but I marked it as deprecated. Need to make sure its documented well in the release.